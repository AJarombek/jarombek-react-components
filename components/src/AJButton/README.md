### Overview

Reusable button component of three different styles (`contained`, `outlined`, and `text`).

**Use cases:**

```javascript
import { AJButton } from 'jarombek-react-components';

const CustomComponent = () => 
  <AJButton type="contained" children="Contained Button" disabled={false} />
```

### Files

| Filename                   | Description                                                                      |
|----------------------------|----------------------------------------------------------------------------------|
| `AJButton.js`              | The reusable `AJButton` component.                                               |
| `index.js`                 | Index file which exports the `AJButton` component.                               |
| `styles.js`                | JSS styles for the `AJButton` component.                                         |
