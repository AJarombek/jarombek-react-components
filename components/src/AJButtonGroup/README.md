### Overview

A group of buttons with specific styles that join them together.

**Use cases:**

```javascript
import { AJButtonGroup, AJButton } from 'jarombek-react-components';

const CustomComponent = () => 
  <AJButtonGroup type="contained">
    <AJButton type="contained">Button #1</AJButton>
    <AJButton type="contained">Button #2</AJButton>
    <AJButton type="contained">Button #3</AJButton>
  </AJButtonGroup>
```

### Files

| Filename                   | Description                                                                           |
|----------------------------|---------------------------------------------------------------------------------------|
| `AJButtonGroup.js`         | The reusable `AJButtonGroup` component.                                               |
| `index.js`                 | Index file which exports the `AJButtonGroup` component.                               |
| `styles.js`                | JSS styles for the `AJButtonGroup` component.                                         |
