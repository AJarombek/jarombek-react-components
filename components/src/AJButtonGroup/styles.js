/**
 * Stylesheet for the AJButtonGroup component using JSS.
 * Author: Andrew Jarombek
 * Date: 10/23/2020
 */

export default {
  buttonGroup: {
    '& > div:not(:first-child):not(:last-child)': {
      borderRadius: 0,
    },
    '& > div:nth-child(n + 1)': {
      marginLeft: '2px',
    },
    '& > div:first-child': {
      borderRadius: '4px 0 0 4px',
    },
    '& > div:last-child': {
      borderRadius: '0 4px 4px 0',
    },
  },
};
