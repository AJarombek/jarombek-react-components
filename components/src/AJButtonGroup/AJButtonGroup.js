/**
 * Grouping of AJButton components.
 * @author Andrew Jarombek
 * @since 10/23/2020
 */

import React from 'react';
import styles from './styles';
import PropTypes from 'prop-types';
import { createUseStyles } from 'react-jss';
import classnames from 'classnames';

const useStyles = createUseStyles(styles);

const AJButtonGroup = ({ children, className }) => {
  const classes = useStyles();

  return <div className={classnames(classes.buttonGroup, className)}>{children}</div>;
};

AJButtonGroup.propTypes = {
  className: PropTypes.string,
};

export default AJButtonGroup;
