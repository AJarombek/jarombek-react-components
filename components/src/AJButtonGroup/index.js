import AJButtonGroup from './AJButtonGroup';

export default AJButtonGroup;
