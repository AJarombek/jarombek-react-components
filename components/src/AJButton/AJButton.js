/**
 * Wrapper component for buttons of multiple different types.  AJButton can be used instead of
 * AJContainedButton, AJOutlinedButton, and AJTextButton.
 * @author Andrew Jarombek
 * @since 1/12/2020
 */

import React from 'react';
import PropTypes from 'prop-types';
import {AJContainedButton, AJOutlinedButton, AJTextButton} from '../../src';

/**
 * Component for a button which can be one of many different types.
 * @param type Whether the button is contained, outlined, or only text.
 * @param children The contents of the button.
 * @param onClick An action that occurs when the button is clicked.
 * @param disabled Whether the button is clickable or not.
 * @return {*} React elements representing a contained, outlined, or text button.
 */
const AJButton = ({type, children, onClick, disabled=false}) => {

  switch (type) {
    case 'contained':
      return (
        <AJContainedButton
          onClick={onClick}
          disabled={disabled}>

          {children}
        </AJContainedButton>
      );
    case 'outlined':
      return (
        <AJOutlinedButton
          onClick={onClick}
          disabled={disabled}>

          {children}
        </AJOutlinedButton>
      );
    case 'type':
      return (
        <AJTextButton
          onClick={onClick}
          disabled={disabled}>

          {children}
        </AJTextButton>
      );
    default:
      return null;
  }
};

AJButton.propTypes = {
  type: PropTypes.oneOf(['contained', 'outlined', 'text']).isRequired,
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func,
  disabled: PropTypes.bool
};

export default AJButton;
