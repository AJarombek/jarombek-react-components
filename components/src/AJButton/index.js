import AJButton from './AJButton';

export default AJButton;
