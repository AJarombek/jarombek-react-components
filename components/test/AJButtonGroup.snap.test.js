/**
 * Snapshot test for the {@link ../../components/AJButtonGroup} component.
 * @author Andrew Jarombek
 * @since 10/24/2020
 */

import React from 'react';
import AJButton from '../src/AJButton/AJButton';
import renderer from 'react-test-renderer';
import AJButtonGroup from "../src/AJButtonGroup/AJButtonGroup";

it('renders correctly', () => {
    const tree = renderer.create(
        <AJButtonGroup>
            <AJButton type="contained" children={<div>Snapshot Button #1</div>} />
            <AJButton type="outlined">
                Snapshot Button #2
            </AJButton>
            <AJButton type="outlined" children={<div>Snapshot Button #3</div>} onClick={f=>f} disabled={true} />
        </AJButtonGroup>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
