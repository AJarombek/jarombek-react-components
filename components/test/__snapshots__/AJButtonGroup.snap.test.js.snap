// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="buttonGroup-0-2-1"
>
  <div
    className="ajBaseButton-0-2-5 ajBaseButtonEnabled-0-2-6 aj-contained-button ajContainedButton-0-2-2 ajContainedButtonEnabled-0-2-3"
  >
    <button
      disabled={false}
      type="button"
    >
      <div>
        Snapshot Button #1
      </div>
    </button>
  </div>
  <div
    className="ajBaseButton-0-2-5 ajBaseButtonEnabled-0-2-6 aj-outlined-button ajOutlinedButton-0-2-8 ajOutlinedButtonEnabled-0-2-9"
  >
    <button
      disabled={false}
      type="button"
    >
      Snapshot Button #2
    </button>
  </div>
  <div
    className="ajBaseButton-0-2-5 ajBaseButtonDisabled-0-2-7 aj-outlined-button ajOutlinedButton-0-2-8 ajOutlinedButtonDisabled-0-2-10"
    onClick={null}
  >
    <button
      disabled={true}
      type="button"
    >
      <div>
        Snapshot Button #3
      </div>
    </button>
  </div>
</div>
`;
